dvc/proc/manager.py

```diff
@@ -85,8 +88,21 @@ def remove(self, name: str, force: bool = False):
             ProcessNotTerminatedError if the specified process is still
             running and was not forcefully killed.
         """
-        raise NotImplementedError
+        process_info = self[name]
+        if process_info.returncode is not None:
+            if not force:
+                raise ProcessNotTerminatedError(name)
+            self.kill(name)
+        remove(os.path.join(self.wdir, name))
 
     def cleanup(self):
         """Remove stale (terminated) processes from this manager."""
-        raise NotImplementedError
+        if not os.path.exists(self.wdir):
+            return
+        for name in os.listdir(self.wdir):
+            try:
+                process_info = self[name]
+            except KeyError:
+                continue
+            if process_info.returncode is not None:
+                remove(os.path.join(self.wdir, name))
```

Comment:
```suggestion
        for process_info in self.processes():
            if process_info.returncode is not None:
                remove(os.path.join(self.wdir, name))
```

No reason to duplicate the iterator exception handling when we can just use `self.processes()`

Reply:
`cleanup` has no name here?

Reply:
Currently, the self.processes() don't provide name info. We have several choices:
1. add `name` to ProcessInfo. 
2. return a (name, ProcessInfo) in `processes()` .
3. return `name`/`key` list in `__iter__(self)`.

For myself, I'd more prefer `3`.

Reply:
3 is probably the best option here (see also https://github.com/iterative/dvc/pull/7048/files#r756744379)

Reply:
> `cleanup` is at the manager level, and should remove all stale processes. To remove individual named processes, the caller should just use `remove()`

Reply:
One thing I was thinking is that eventually `cleanup` may need a `force` parameter, to remove all processes whether or not they are still running, but that's not a requirement for now.